### PR TITLE
Implemented bogosort

### PR DESCRIPTION
**Pull request by lord-123**
I believe I have implemented issue #1. This is my first time using Go however so this might not be the right way to do it. It simply shuffles the array until it is sorted (as can be quite easily seen in the code).

### CODE REVIEW

**Review by aod — CHANGES_REQUESTED** [review]
This looks good. I left a few comments that should be addressed before this gets merged. Furthermore I haven't documented this anywhere but could you run `gofmt` for formatting code and `goimports` for managing imports? I can also do this for you if you don't want to, since it's my fault for not specifying this.

File: pkg/sorters/bogo.go [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+			update <- i
+			update <- j
+		})
+
+		sorted := true
+		for i := 0; i < n - 1; i++ {
+			if arr[i] > arr[i + 1] {
+				sorted = false
+			}
+		}
```
Comment (aod) [inline pkg/sorters/bogo.go]: Can you do an early break out of the loop if `sorted` is false. For example if the first two are not sorted we don't want to go through the rest of the elements. Instead it should immediately go back to shuffling.

File: pkg/sorters/bogo.go [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+		if sorted == true {
```
Comment (aod) [inline pkg/sorters/bogo.go]: Since it's a `bool` type you don't need the comparison operator

File: pkg/sorters/bogo.go [hunk 3]
```diff
@@ -0,0 +1,29 @@
+package sorters
+
+import "sync"
+import "math/rand"
```
Comment (aod) [inline pkg/sorters/bogo.go]: Combine imports:
```go
import (
	"math/rand"
	"sync"
)
```

**Review by aod — CHANGES_REQUESTED** [review]
Whoops, looks like I messed up the previous review. I didn't actually start the review which removed my comments. This time it should be clear on what needs to be changed before merging. Sorry about that. Whatever was said in the previous one still remains fyi.

**Comment by lord-123** [comment]
I'm not sure how to use `gofmt` and `goimports` so I'm not sure if this is quite up to spec yet but I have added the adjustments you asked for.

**Comment by aod** [comment]
LGTM!

Thank you for your contribution :+1:

**Pull request merged**